pkgs/development/tools/godot/mono/default.nix

```diff
@@ -0,0 +1,94 @@
+{ godot
+, lib
+, callPackage
+, mkNugetDeps
+, mkNugetSource
+, mono
+, dotnet-sdk
+, writeText
+}:
+
+godot.overrideAttrs (self: base: {
+  pname = "godot-mono";
+
+  passthru = let bp = base.passthru; in bp // {
+    sconsFlagSet = bp.sconsFlagSet // {
+      module_mono_enabled = true;
+      mono_prefix = mono;
+    };
+
+    buildDescription = "mono build";
+
+    glue = callPackage ./glue.nix {};
+
+    make-deps = callPackage ./make-deps.nix {};
+  };
+
+  nativeBuildInputs = base.nativeBuildInputs ++ [ mono dotnet-sdk ];
+
+  nugetDeps = mkNugetDeps { name = "deps"; nugetDeps = import ./deps.nix; };
+
+  nugetSource =
+    mkNugetSource {
+      name = "${self.pname}-nuget-source";
+      description = "A Nuget source with dependencies for ${self.pname}";
+      deps = [ self.nugetDeps ];
+    };
```

Comment:
```suggestion
  nugetSource = mkNugetSource {
    name = "${self.pname}-nuget-source";
    description = "A Nuget source with dependencies for ${self.pname}";
    deps = [ (mkNugetDeps { name = "deps"; nugetDeps = import ./deps.nix; }) ];
  };
```

Reply:
I initially had that, but then decided it might be helpful to be able to see the nugetDeps from nix-shell.  Making it an attribute of its own makes `$nugetDeps` refer to its path.

Reply:
I found one other user of mkNugetSource and that one hides the variable in a let in prefixed with an underscore which means it should be treated privately https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/writers/default.nix#L309-L315

Reply:
Why does the fact that another derivation is using it privately mean it *should* be used privately.  (Also, by "privately" I'm assuming you mean "not shared with the phases as environment variables".)

In any case, `nugetDeps` and `nugetSource` are not needed by the phases.  The only reason I left them in as attributes was for the convenience of _seeing_ them.  However, a better way to do this is passthru, which accomplishes the "privacy" of a let, while still being overridable and accessible to someone who wants to see what they contain.  I'll make this change.